### PR TITLE
Support/wagtail42

### PR DESCRIPTION
**Pull request by nickmoreton**
This updates the package for wagtail 4.2 support.

Adds:
- Testing for wagtail >= 4.1

Removes:
- Support for wagtail < 4.1
- Testing for wagtail < 4.1

### CODE REVIEW

**Comment by cnk** [comment]
Thanks @nickmoreton! Once I get the release done for versions <= 3.0 I will merge this. 

I am holding off on that release in hopes that someone can figure out #32 which affects all versions of wagtail-hallo. Then we can merge this PR and the one that fixes the document and image choosers for Wagtail 4+ #33

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -46,31 +46,30 @@ jobs:
...
             django: '4.1'
-            python: '3.10'
+            wagtail: '4.2'
+
+          - python: '3.11'
+            django: '3.2'
+            wagtail: '4.1'
+          - python: '3.11'
+            django: '3.2'
+            wagtail: '4.2'
```
Comment (katdom13) [inline .github/workflows/ci.yml:72]: I don't think there's a need to exclude this if the db is sqlite

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -91,39 +90,31 @@ jobs:
...
-            postgres: '10.8'
-            python: '3.10'
+            wagtail: '4.2'
+
+          - python: '3.11'
+            django: '3.2'
+            wagtail: '4.1'
+          - python: '3.11'
+            django: '3.2'
+            wagtail: '4.2'
```
Comment (katdom13) [inline .github/workflows/ci.yml:117]: These could be tested for postgres if you add postgres 15:
```
- python: '3.11'
   django: '3.2'
   wagtail: '4.1'
   postgres: '15'
```

File: tox.ini [hunk 3]
```diff
@@ -3,9 +3,9 @@ skipsdist = True
 usedevelop = True
 
 envlist =
-    python{3.7,3.8,3.9,3.10}-django{3.2}-wagtail{2.15,2.16,3.0,4.0,main}-{sqlite,postgres}
-    python{3.8,3.9,3.10}-django{4.0}-wagtail{2.16,3.0,4.0,main}-{sqlite,postgres}
-    python{3.8,3.9,3.10}-django{4.1,main}-wagtail{4.0,main}-{sqlite,postgres}
+    python{3.7}-django{3.2}-wagtail{4.1,4.2,main}-{sqlite,postgres}
+    python{3.8,3.9,3.10}-django{4.0,4.1}-wagtail{4.1,4.2,main}-{sqlite,postgres}
+    python{3.11}-django{4.1,main}-wagtail{4.1,4.2,main}-{sqlite,postgres}
```
Comment (katdom13) [inline tox.ini:8]: Suggestion:
- Create two postgres environments
- Test different combinations of python, django and wagtail

```
    python{3.7}-django{3.2}-wagtail{4.1,4.2,main}-{sqlite,postgres12}
    python{3.8,3.9,3.10,3.11}-django{3.2,4.0,4.1,main}-wagtail{4.1,4.2,main}-sqlite
    python{3.8,3.9,3.10}-wagtail{4.1,4.2,main}-{postgres12,postgres15}
    python{3.8,3.9,3.10}-wagtail{3.2}-postgres12
    python{3.11}-wagtail{4.1,4.2,main}-postgres15

deps =
    ...

    postgres12: psycopg2>=2.6,<2.9.5
    postgres15: psycopg2>=2.9.5
```
As suggested from this conversation: https://torchbox.slack.com/archives/C03R0FNK7/p1675655322697539
Reference: https://github.com/wagtail/wagtail-localize/pull/673/files#r1097252879

Comment (katdom13) [reply tox.ini:8]: Apparently, newer versions of postgres (15+) work with Python 3.11 but not with older Django (3.2); older versions of postgres (10 - 12) do not work with Python 3.11 but work with Django 3.2.
So tests with Django 3.2 are isolated.

**Comment by katdom13** [comment]
Hi @nickmoreton ,
Left a few comments for your consideration, but this LGTM.
Thanks!

**Comment by jozo** [comment]
Hi, I've forked code from #39 and (hopefully) fixed #33 . Can you please check it and merge it? @cnk 
It's in branch `fix-toolbar-in-wagtail4` - https://github.com/jozo/wagtail-hallo/tree/fix-toolbar-in-wagtail4

**Comment by jorenham** [comment]
@nickmoreton why did you delete your fork?

**Comment by nickmoreton** [comment]
> @nickmoreton why did you delete your fork?

Ah that was me doing some house keeping. I was doing this work as part of a larger project including many other package updates. We're now using a new organisation on GitHub to carry out the work. We're updating packages used on client sites and this was no longer used by us. I will see if i can get a local copy back in place and possibly set it up again if it's something you need. I am out of the office till next week but will have a look asap 
Sorry if this has caused problems for you. 😕

**Comment by jorenham** [comment]
@nickmoreton no worries, see #40

**Comment by nickmoreton** [comment]
@jorenham I do have a local copy still if you need anything form it but it looks like your PR includes the changes. 👍

**Pull request closed without merging**